<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>965</width>
    <height>583</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QVBoxLayout" name="mainLayout">
      <item>
       <widget class="VidLabel" name="vidLabel">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="baseSize">
         <size>
          <width>768</width>
          <height>432</height>
         </size>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="trackBarSlider">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QPushButton" name="playpauseButton">
          <property name="text">
           <string>Play</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="timeLabel">
          <property name="text">
           <string>--:--/--:--</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="BuyButton">
          <property name="text">
           <string>Buy Product</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="showObjectsButton">
          <property name="text">
           <string>Show Products</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="showSideBarButton">
          <property name="text">
           <string>Show Sidebar</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QWidget" name="objectSideBar" native="true">
      <property name="maximumSize">
       <size>
        <width>250</width>
        <height>16777215</height>
       </size>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="ObjectListWidget" name="objectList"/>
       </item>
       <item>
        <widget class="QPushButton" name="createObjectButton">
         <property name="text">
          <string>Create Object</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QPushButton" name="moveRectButton">
           <property name="text">
            <string>Move Rect</string>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="editObjectButton">
           <property name="text">
            <string>Edit Object</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>965</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave_Objects"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionSave_Objects">
   <property name="text">
    <string>Save Objects</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>VidLabel</class>
   <extends>QLabel</extends>
   <header>vidlabel.h</header>
  </customwidget>
  <customwidget>
   <class>ObjectListWidget</class>
   <extends>QListWidget</extends>
   <header>objectlistwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>playpauseButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onPlay()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>48</x>
     <y>560</y>
    </hint>
    <hint type="destinationlabel">
     <x>482</x>
     <y>291</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>showSideBarButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>objectSideBar</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>440</x>
     <y>560</y>
    </hint>
    <hint type="destinationlabel">
     <x>720</x>
     <y>301</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>trackBarSlider</sender>
   <signal>sliderPressed()</signal>
   <receiver>MainWindow</receiver>
   <slot>trackBarSliderPressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>354</x>
     <y>532</y>
    </hint>
    <hint type="destinationlabel">
     <x>482</x>
     <y>291</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>trackBarSlider</sender>
   <signal>sliderReleased()</signal>
   <receiver>MainWindow</receiver>
   <slot>trackBarSliderReleased()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>354</x>
     <y>532</y>
    </hint>
    <hint type="destinationlabel">
     <x>482</x>
     <y>291</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>createObjectButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>setMouseCallbackAddObject(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>830</x>
     <y>522</y>
    </hint>
    <hint type="destinationlabel">
     <x>482</x>
     <y>291</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>moveRectButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>setMouseCallbackMoveRect(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>752</x>
     <y>552</y>
    </hint>
    <hint type="destinationlabel">
     <x>482</x>
     <y>291</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>objectList</sender>
   <signal>doubleClicked(QModelIndex)</signal>
   <receiver>MainWindow</receiver>
   <slot>onItemDoubleClick(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>830</x>
     <y>271</y>
    </hint>
    <hint type="destinationlabel">
     <x>482</x>
     <y>291</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>showObjectsButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>showObjects(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>577</x>
     <y>560</y>
    </hint>
    <hint type="destinationlabel">
     <x>482</x>
     <y>291</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>editObjectButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>showEditObjectDialog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>889</x>
     <y>552</y>
    </hint>
    <hint type="destinationlabel">
     <x>482</x>
     <y>291</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onPlay()</slot>
  <slot>trackBarSliderPressed()</slot>
  <slot>trackBarSliderReleased()</slot>
  <slot>setMouseCallbackAddObject(bool)</slot>
  <slot>setMouseCallbackMoveRect(bool)</slot>
  <slot>showEditObjectDialog()</slot>
  <slot>onItemDoubleClick(QModelIndex)</slot>
  <slot>showObjects(bool)</slot>
 </slots>
</ui>
